api/filters/paging.go

```diff
@@ -52,22 +52,32 @@ func (*pagingFilter) Name() string {
 func (pf *pagingFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
 	ctx := req.Context()
 	maxItems := req.URL.Query().Get("max_items")
-	limit, err := pf.validateMaxItemsQuery(maxItems)
+	limit, err := pf.parseMaxItemsQuery(maxItems)
 	if err != nil {
 		return nil, err
 	}
-	ctx = context.WithValue(ctx, "limit", limit)
+
+	rawToken := req.URL.Query().Get("token")
+	token, err := pf.parsePageToken(rawToken)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = web.ContextWithPageLimit(ctx, limit)
 	if limit > 0 {
-		ctx, err = query.AddCriteria(ctx, query.LimitResultBy(limit))
+		ctx, err = query.AddCriteria(ctx, query.LimitResultBy(limit+1),
+			query.OrderResultBy("paging_sequence", query.AscOrder),
+			query.ByField(query.GreaterThanOperator, "paging_sequence", strconv.Itoa(token)))
```

Comment:
Atoi in parse method and Itoa here  probably cancel out - you could directly use the string but we should check how meaningful the error message will be